### PR TITLE
feat(search): Use ServiceDelegator to run both backends

### PR DESCRIPTION
**Pull request by iProgramStuff**

### CODE REVIEW

File: src/sentry/search/base.py [hunk 1]
```diff
@@ -26,3 +31,60 @@ def query(
...
     ):
         raise NotImplementedError
+
+
+def selector_func(context, method, callargs):
+    backends = ["default"]
+    if settings.SENTRY_USE_MORESNUBA:
+        backends.append("moresnuba")
+
+    return backends
```
Comment (fpacifici) [inline src/sentry/search/base.py]: I don't think this is enough. Unless I am missing something, here we are trying to build a three way choice:
1) whether to go for the old system or the new system, and this is driven by USE_MORE_SNUBA
2) if we go for the new system, depending on the query we have to go for postgres + snuba or snuba only.

I think this approach is missing the second choice and always go for a full snuba search  if SENTRY_USE_MORESNUBA is true.

Comment (iProgramStuff) [reply src/sentry/search/base.py]: It wasn't my intention to support a 3 way choice here - it either runs the original, as it always has, or if USE_MORESNUBA is true, it will run both and log the results for analysis.

I can add in an option to do the MORESNUBA queries only though, as that is simple enough.

Comment (fpacifici) [reply src/sentry/search/base.py]: Sorry, let me rephrase, I did not mean that we have to implement the three way choice in that method. I mean that the overall logic now is missing the second choice, if USE_MORESNUBA is True then it will ignore postgres no matter what the query is,  which will yield the wrong results for those queries that cannot be managed entirely by Snuba.
Doing that will skew the results of your comparison since the error rate will be very high.
In the MoreSnuba backend, there has to exist a code path for postgres + snuba which depends on the query parameters.

Now the interesting question to answer is:
when USE_MORESNUBA is true, if the query still requires postgres, is the resulting query the same as if USE_MORESNUBA was false (except for the dataset name) or is it a brand new query? In the first case we should try to reuse the logic of the old backend for that instead of rewriting it from scratch.
